
project frameworks/base/
diff --git a/frameworks/base/services/core/java/com/android/server/display/ColorFade.java b/frameworks/base/services/core/java/com/android/server/display/ColorFade.java
index 900133c..333baa5 100644
--- a/frameworks/base/services/core/java/com/android/server/display/ColorFade.java
+++ b/frameworks/base/services/core/java/com/android/server/display/ColorFade.java
@@ -450,6 +450,7 @@ final class ColorFade {
         if (!attachEglContext()) {
             return false;
         }
+        SurfaceTexture st = null;
         try {
             if (!mTexNamesGenerated) {
                 GLES20.glGenTextures(1, mTexNames, 0);
@@ -459,7 +460,7 @@ final class ColorFade {
                 mTexNamesGenerated = true;
             }
 
-            final SurfaceTexture st = new SurfaceTexture(mTexNames[0]);
+            st = new SurfaceTexture(mTexNames[0]);
             final Surface s = new Surface(st);
             try {
                 SurfaceControl.screenshot(SurfaceControl.getBuiltInDisplay(
@@ -484,6 +485,7 @@ final class ColorFade {
             GLES20.glViewport(0, 0, mDisplayWidth, mDisplayHeight);
             ortho(0, mDisplayWidth, 0, mDisplayHeight, -1, 1);
         } finally {
+            if (st != null) st.release();
             detachEglContext();
         }
         return true;
