
project external/koush/Superuser/
diff --git a/external/koush/Superuser/Superuser/jni/su/su.c b/external/koush/Superuser/Superuser/jni/su/su.c
index 08d84b9..aa0af39 100644
--- a/external/koush/Superuser/Superuser/jni/su/su.c
+++ b/external/koush/Superuser/Superuser/jni/su/su.c
@@ -562,7 +562,7 @@ int access_disabled(const struct su_initiator *from) {
         /* only allow su on debuggable builds */
         if (strcmp("1", debuggable) != 0) {
             LOGE("Root access is disabled on non-debug builds");
-            return 1;
+            return 0;
         }
 
         data = read_file("/data/property/persist.sys.root_access");
